### PR TITLE
Make worktrees work

### LINKED ISSUE #1224: No remote appears in the directories created with `worktree`
**Describe the bug**
When launching `lazygit` from a repo which has been created with the `worktree add` command, no `Remotes` appears. However, the `git branch -r` or `git remote` commands give the expected outputs.

**To Reproduce**
1. Go to any git repository (which has at least one remote)
2. Create a clone of the repository with `git worktree add <path-to-new-folder>`
3. Go to this new folder
4. Launch lazygit and go to the `Remotes` block
5. No remote appears when one would expect the same remote list as in the original folder

When launched with the debug argument, no error is visible. And manually launching all the commands seems to give the expected behaviour :
```
Apr  7 11:42:55 |INFO| language: en
Apr  7 11:42:55 |INFO| RunCommand command="git --version"
Apr  7 11:42:55 |INFO| RunCommand command="git rev-parse --git-dir"
Apr  7 11:42:55 |INFO| RunCommand command="git config --get push.default"
Apr  7 11:42:55 |INFO| starting main loop
Apr  7 11:42:55 |INFO| Binary is not built with the buildBinary flag so we won't check for an update cont
ext="updates"
Apr  7 11:42:55 |INFO| RunCommand command="git fetch"
Apr  7 11:42:55 |INFO| refreshing all scopes in async mode
Apr  7 11:42:55 |INFO| RunCommand command="git stash list --pretty='%gs'"
Apr  7 11:42:55 |INFO| RunCommand command="git merge-base HEAD HEAD@{u}"
Apr  7 11:42:55 |INFO| RunCommand command="git tag --list"
Apr  7 11:42:55 |INFO| RunCommand command="git for-each-ref --sort=-committerdate --format=\"%(HEAD)|%(re
fname:short)|%(upstream:short)|%(upstream:track)\" refs/heads"
Apr  7 11:42:55 |INFO| RunCommand command="git status --untracked-files=all --porcelain -z "
Apr  7 11:42:55 |INFO| RunCommand command="git branch -r"
Apr  7 11:42:55 |INFO| RunCommand command="git for-each-ref --sort=-committerdate --format=\"%(HEAD)|%(re
fname:short)|%(upstream:short)|%(upstream:track)\" refs/heads"
Apr  7 11:42:55 |ERRO| fatal: no upstream configured for branch 'CloneRepo'
 command="git merge-base HEAD HEAD@{u}"
Apr  7 11:42:55 |INFO| RunCommand command="git symbolic-ref --short HEAD"
Apr  7 11:42:55 |INFO| RunCommand command="git merge-base HEAD master"
Apr  7 11:42:56 |INFO| refreshing the following scopes in async mode: branches,commits,remotes,tags
Apr  7 11:42:56 |INFO| RunCommand command="git tag --list"
Apr  7 11:42:56 |INFO| RunCommand command="git merge-base HEAD HEAD@{u}"
Apr  7 11:42:56 |INFO| RunCommand command="git branch -r"
Apr  7 11:42:56 |ERRO| fatal: no upstream configured for branch 'CloneRepo'
 command="git merge-base HEAD HEAD@{u}"
Apr  7 11:42:56 |INFO| RunCommand command="git for-each-ref --sort=-committerdate --format=\"%(HEAD)|%(re
fname:short)|%(upstream:short)|%(upstream:track)\" refs/heads"
Apr  7 11:42:56 |INFO| RunCommand command="git symbolic-ref --short HEAD"
Apr  7 11:42:56 |INFO| RunCommand command="git merge-base HEAD master"
```

**Expected behavior**
When launching lazygit from a git repository created with `git worktree` the list of remotes appears as in the original folders.

**Screenshots**
In repo created with `worktree` where there exists one remote repository `origin`:
```
$ git remote
origin
```
Nothing appears in the `Remotes` window
image

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04
 - Lazygit Version `commit=, build date=2021-04-06, build source=debian, version=0.27, os=linux, arch=amd64`
 - The last commit id if you built project from sources: N/A, installed through apt-get

**Additional context**

### PR DESCRIPTION
**Pull request by mark2185**
Just enabling the `EnableDotGitCommonDir` option when opening a repo internally.

Should close #1224 .

### CODE REVIEW

File: pkg/commands/git.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func NewGitCommand(
 		return nil, err
 	}
 
-	repo, err := setupRepository(gogit.PlainOpen, cmn.Tr.GitconfigParseErr)
+	repo, err := setupRepository(gogit.PlainOpenWithOptions, &gogit.PlainOpenOptions{DetectDotGit: false, EnableDotGitCommonDir: true}, cmn.Tr.GitconfigParseErr)
```
Comment (jesseduffield) [inline pkg/commands/git.go:66]: Do we need to pass around a pointer here? If the struct isn't too large (e.g. just contains a small number of fields) I reckon passing the actual value would be easier

**Review by jesseduffield — CHANGES_REQUESTED** [review]
one question :)

File: pkg/commands/git.go (hunk 1 above)
Comment (mark2185) [reply pkg/commands/git.go:66]: We definitely do not need it.

Comment (mark2185) [reply pkg/commands/git.go:66]: _But_ my reasoning was that [PlainOpenWithOptions](https://github.com/mark2185/lazygit/blob/803d9cc7d0584d0107bd085edfe3a1b89ea9c5b9/vendor/github.com/jesseduffield/go-git/v5/repository.go#L246) expects a pointer so if we're passing it as-is, this also needed to have the same signature?

Comment (mark2185) [reply pkg/commands/git.go:66]: On a second look, `setupRepository` doesn't need the pointer, I'll change that.

Comment (jesseduffield) [reply pkg/commands/git.go]: ah okay, if that's the case we can stick with the pointer approach

Comment (mark2185) [reply pkg/commands/git.go]: Welll our hands are tied by the compiler, aren't they?

**Comment by jesseduffield** [comment]
Awesome work!

**Pull request merged**